pkg/apis/networking/validation/validation_test.go

```diff
@@ -1566,267 +1566,243 @@ func TestValidateIngressUpdate(t *testing.T) {
 	}
 }
 
+type netIngressTweak func(ingressClass *networking.IngressClass)
+
+func makeValidIngressClass() *networking.IngressClass {
+	return &networking.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test123",
+		},
+		Spec: networking.IngressClassSpec{
+			Controller: "foo.co/bar",
+		},
+	}
+}
+
+func makeValidIngressClassUpdateCustom(tweaks ...netIngressTweak) *networking.IngressClass {
+	ingressClass := makeValidIngressClass()
+	for _, fn := range tweaks {
+		fn(ingressClass)
+	}
+	return ingressClass
+}
+
 func TestValidateIngressClass(t *testing.T) {
+	setIngressClassObjectMetaName := func(name string) netIngressTweak {
+		return func(ingressClass *networking.IngressClass) {
+			ingressClass.ObjectMeta.Name = name
+		}
+	}
+
+	setIngressClassSpecController := func(controller string) netIngressTweak {
+		return func(ingressClass *networking.IngressClass) {
+			ingressClass.Spec.Controller = controller
+		}
+	}
+
+	setIngressClassSpecAddParameters := func(ingressClass *networking.IngressClass) {
+		ingressClass.Spec.Parameters = &networking.IngressClassParametersReference{}
+	}
+
+	setIngressClassSpecAddParametersAPIGroup := func(apiGroup *string) netIngressTweak {
+		return func(ingressClass *networking.IngressClass) {
```

Comment:
Rather than all these funcs, I'd just make one that takes the args and builds.  E.g.

```
Func makeIngressClassParams(group, kind, scope namespace, name) networking.IngressClassParams {
}

Func setParams(params networking.IngressClassParams) tweak {
    Return func(I) {
        I.spec.params = params
    }
}
```

(Sorry for typos, on mobile)
```

Reply:
Done